### ISSUE TITLE
Seeing a lot of API MISUSE messages in the console while running instabug network logging

### ISSUE THREAD

**Issue by BrentMifsud** [Instabug/Instabug-iOS#443]
### Steps to Reproduce the Problem
Enabled instabug network logging

### Expected Behavior
Should not be seeing console log messages
 
### Actual Behavior
```
2021-12-09 14:25:47.141795-0500 <MyApp>[2673:875842] API MISUSE: NSURLSession delegate IBGURLSessionDemux: <IBGURLSessionDemux: 0x281b2f4e0> (0x281b2f4e0)
2021-12-09 14:25:47.143581-0500 <MyApp>[2673:875842] API MISUSE: task:willPerformHTTPRedirection:newRequest:completionHandler: completion handler not called
2021-12-09 14:25:57.150380-0500 <MyApp>[2673:875842] API MISUSE: NSURLSession delegate IBGURLSessionDemux: <IBGURLSessionDemux: 0x281b2f4e0> (0x281b2f4e0)
2021-12-09 14:25:57.150770-0500 <MyApp>[2673:875842] API MISUSE: task:willPerformHTTPRedirection:newRequest:completionHandler: completion handler not called
2021-12-09 14:26:07.160213-0500 <MyApp>[2673:875842] API MISUSE: NSURLSession delegate IBGURLSessionDemux: <IBGURLSessionDemux: 0x281b2f4e0> (0x281b2f4e0)
2021-12-09 14:26:07.161453-0500 <MyApp>[2673:875842] API MISUSE: task:willPerformHTTPRedirection:newRequest:completionHandler: completion handler not called
```

### Instabug integration code
```swift
    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
        setupInstabug()
        if let viewController = window?.rootViewController as? RootViewController {
            let baseURL = Environment.current.baseURL
            let apiToken = Environment.current.apiToken
            let urlSessionConfig = URLSessionConfiguration.default
            urlSessionConfig.timeoutIntervalForResource = 10

            // Prevent network cache from growing out of control
            // https://stackoverflow.com/questions/1330416/cfnetwork-memory-leaks-in-iphone
            urlSessionConfig.urlCache = URLCache(memoryCapacity: 0, diskCapacity: 0, diskPath: nil)

            // Adds session to Instabug logger
            NetworkLogger.enableLogging(for: urlSessionConfig)
            
           // Some unrelated code...
        }

        // Some unrelated code...

        // Manually invoke Instabug when detecting bug report gesture
        instabugGestureObserver = NotificationCenter.default.addObserver(forName: .onBugReport, object: nil, queue: .main) { _ in
            Instabug.show()
        }

        return true
    }
```

```swift
    func setupInstabug() {
        // set up instabug
        Instabug.start(withToken: Constants.instabugToken, invocationEvents: [.none])
        Instabug.welcomeMessageMode = .disabled
        BugReporting.promptOptionsEnabledReportTypes = [.bug, .feedback]
        Replies.enabled = false
        // Also uploading the log because it goes beyond the last session
        Instabug.willSendReportHandler = { report in
            DDLog.flushLog()
            if let deviceId = UIDevice.current.identifierForVendor?.uuidString {
                Instabug.setUserAttribute(deviceId, withKey: "Device ID")
            }
            report.addFileAttachment(with: URL(fileURLWithPath: self.fileLogger?.currentLogFileInfo?.filePath ?? ""))
            if let logs = self.fileLogger?.logFileManager.sortedLogFilePaths, logs.count > 1 {
                Instabug.addFileAttachment(with: URL(fileURLWithPath: logs[1]))
            }
            return report
        }
        self.isInstabugActive = true
    }
```

### SDK Version
10.11.0
### iOS Version
15.1
### Device Model
iPad

**Comment by hassaanelgarem** [comment]
We're looking into this.
Can I ask if the redirected requests are failing or passing?

**Comment by BrentMifsud** [comment]
the redirects are going through.

But I did notice that while network logging is enabled, there seems to be a slight memory leak (about ~2 megabytes per hour).

Our app is an iPad kiosk that needs to run for long periods of time. And this was causing the app to kill itself after a week or so of runtime.

**Comment by hassaanelgarem** [comment]
Can you export the instruments trace and send it please? Or if not possible a screenshot of the leak could also be useful.

**Comment by BrentMifsud** [comment]
> Can you export the instruments trace and send it please? Or if not possible a screenshot of the leak could also be useful.

Yea I can send one over. Id prefer not to post it publicly here though. Is there an email I can forward it to or give you an iCloud link or something?

Here are some screenshots in the meantime:
<img width="1728" alt="Screen Shot 2021-12-11 at 2 56 58 AM" src="https://user-images.githubusercontent.com/18543934/145670464-be5daa8a-836f-465e-96f0-acc6ef6d45bd.png">
<img width="1728" alt="Screen Shot 2021-12-11 at 2 57 59 AM" src="https://user-images.githubusercontent.com/18543934/145670465-71842c32-3950-4c07-89d9-b3986dc0e47a.png">
<img width="1728" alt="Screen Shot 2021-12-11 at 2 58 09 AM" src="https://user-images.githubusercontent.com/18543934/145670467-8d9b01bc-4218-45ec-9d69-b4da65230462.png">
<img width="1728" alt="Screen Shot 2021-12-11 at 2 58 19 AM" src="https://user-images.githubusercontent.com/18543934/145670468-6148628a-9cd6-443b-bf74-88a4b6fa890b.png">

**Comment by BrentMifsud** [comment]
@hassaanelgarem sent you a copy of the trace to the email you have on your GitHub profile.

**Comment by hassaanelgarem** [comment]
@BrentMifsud Thank you! We'll take a look and get back to you soon.

**Comment by BrentMifsud** [comment]
> @BrentMifsud Thank you! We'll take a look and get back to you soon.

Great! thanks for looking into this.

**Comment by husseinkishk** [comment]
Hi @BrentMifsud, Would you please Set this block `-[IBGNetworkLogger setTaskWillPerformHTTPRedirectionBlock:]` if your app attempting to perform a redirection to a different URL and you have network logging enabled.
Regrading the memory leaks, We're still looking into it but could you please reproduce the leaks **after passing your completionHandler to our API** and send us the memory graph of the leaks.

**Comment by BrentMifsud** [comment]
Currently trying to figure out how to export the memory graph for you.

But just an FYI. By adding that completion handler you mentioned above, the memory leak is now gone.

**Comment by husseinkishk** [comment]
@BrentMifsud I've the one you shared before to my colleague. Thank you.

**Comment by BrentMifsud** [comment]
Bumping this to keep it open.

**Comment by ahmedsalah196** [comment]
@BrentMifsud I'm glad to inform you that this issue should be fixed on the latest version 10.11.4. I'd really like to thank you for opening this ticket which helped in increasing our SDK quality 🎉 
I am closing this issue but feel free to reopen it if the issue persists 🙏
